File: javascript/ExampleEnumerate.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;

// Create connection and connect to brickd
ipcon = new Tinkerforge.IPConnection();
ipcon.connect(HOST, PORT);

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Trigger Enumerate
        ipcon.enumerate();
    }
);

// Register Enumerate Callback
ipcon.on(Tinkerforge.IPConnection.CALLBACK_ENUMERATE,
    // Print incoming enumeration
    function(uid, connectedUid, position, hardwareVersion, firmwareVersion,
             deviceIdentifier, enumerationType) {
        console.log('UID:               '+uid);
        console.log('Enumeration Type:  '+enumerationType);
        if(enumerationType === Tinkerforge.IPConnection.ENUMERATION_TYPE_DISCONNECTED) {
            console.log('');
            return;
        }
        console.log('Connected UID:     '+connectedUid);
        console.log('Position:          '+position);
        console.log('Hardware Version:  '+hardwareVersion);
        console.log('Firmware Version:  '+firmwareVersion);
        console.log('Device Identifier: '+deviceIdentifier);
        console.log('');
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: perl/example_enumerate.pl
#!/usr/bin/perl

use Tinkerforge::IPConnection;

use constant HOST => 'localhost';
use constant PORT => 4223;

# Print incoming enumeration
sub cb_enumerate
{
	my ($uid, $connected_uid, $position, $hardware_version,
	    $firmware_version, $device_identifier, $enumeration_type) = @_;

	print "UID:               $uid\n";
	print "Enumeration Type:  $enumeration_type\n";

	if ($enumeration_type == Tinkerforge::IPConnection->ENUMERATION_TYPE_DISCONNECTED)
	{
		print "\n";
		return;
	}

	print "Connected UID:     $connected_uid\n";
	print "Position:          $position\n";
	print "Hardware Version:  ".join('.', @$hardware_version)."\n";
	print "Firmware Version:  ".join('.', @$firmware_version)."\n";
	print "Device Identifier: $device_identifier\n";
	print "\n";
}

# Create connection and connect to brickd
my $ipcon = Tinkerforge::IPConnection->new();

$ipcon->connect(&HOST, &PORT);

# Register Enumerate Callback
$ipcon->register_callback($ipcon->CALLBACK_ENUMERATE, 'cb_enumerate');

# Trigger Enumerate
$ipcon->enumerate();

print "Press any key to exit...\n";
<STDIN>;
$ipcon->disconnect();
